test/csuite/wt8290_rollback_reason/main.c

```diff
@@ -0,0 +1,79 @@
+ * means.
+ *
+ * In jurisdictions that recognize copyright laws, the author or authors
+ * of this software dedicate any and all copyright interest in the
+ * software to the public domain. We make this dedication for the benefit
+ * of the public at large and to the detriment of our heirs and
+ * successors. We intend this dedication to be an overt act of
+ * relinquishment in perpetuity of all present and future rights to this
+ * software under copyright law.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+ * IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+ * OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+ * ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+ * OTHER DEALINGS IN THE SOFTWARE.
+ */
+#include "test_util.h"
+
+const char *rollback_error = "oldest pinned transaction ID rolled back for eviction";
+
+/*
+ * JIRA ticket reference: WT-4699 Test case description: Use a JSON dump cursor on a projection, and
+ * overwrite the projection string. Failure mode: On the first retrieval of a JSON key/value, a
+ * configure parse error is returned.
+ */
+int
+main(int argc, char *argv[])
+{
+    TEST_OPTS *opts, _opts;
+    WT_CURSOR *c;
+    WT_DECL_RET;
+    WT_SESSION *session;
+    const char *rollback_reason, *reason_2;
+    WT_UNUSED(c);
+    opts = &_opts;
+    memset(opts, 0, sizeof(*opts));
+    testutil_check(testutil_parse_opts(argc, argv, opts));
+    testutil_make_work_dir(opts->home);
+
+    testutil_check(wiredtiger_open(opts->home, NULL, "create,cache_size=1M", &opts->conn));
+    testutil_check(opts->conn->open_session(opts->conn, NULL, NULL, &session));
+    testutil_check(session->create(session, opts->uri, "key_format=i,value_format=S"));
+    testutil_check(session->open_cursor(session, opts->uri, NULL, NULL, &c));
+    rollback_reason = session->get_rollback_reason(session);
+    testutil_assert(rollback_reason == NULL);
+
+    testutil_check(session->begin_transaction(session, NULL));
+    for (int i = 0; i < 20000; i++) {
+        c->set_key(c, i);
+        c->set_value(c, "abcdefghijklmnopqrstuvwxyz");
+        ret = c->insert(c);
+        if (ret != 0) {
+            if (ret == WT_ROLLBACK) {
+                reason_2 = session->get_rollback_reason(session);
+                testutil_assert(strcmp(reason_2, rollback_error) == 0);
+                break;
+            } else
+                testutil_die(ret, "Unexpected error occurred while inserting values.");
+        }
+    }
+    testutil_check(session->rollback_transaction(session, NULL));
+    rollback_reason = session->get_rollback_reason(session);
+    testutil_assert(strcmp(rollback_reason, rollback_error) == 0);
+
+    testutil_check(session->close(session, NULL));
+    testutil_cleanup(opts);
+    return (EXIT_SUCCESS);
+}
```

Comment:
Since you're adding a method in `wiredtiger.in`, you' d actually have to do additional work to avoid the method showing up in Python, so might as well use that instead of writing a standalone test?

I think this:
```
diff --git a/lang/python/wiredtiger.i b/lang/python/wiredtiger.i
index c77a2effd..3bcaa9997 100644
--- a/lang/python/wiredtiger.i
+++ b/lang/python/wiredtiger.i
@@ -594,6 +594,7 @@ COMPARE_NOTFOUND_OK(__wt_cursor::_search_near)
 %exception __wt_connection::get_home;
 %exception __wt_connection::is_new;
 %exception __wt_connection::search_near;
+%exception __wt_session::get_rollback_reason;
 %exception __wt_cursor::_set_key;
 %exception __wt_cursor::_set_key_str;
 %exception __wt_cursor::_set_value;
```
is all you need.

Reply:
Interesting I wasn't expecting a string return function to work automatically in python. I won't exclude it in that case.

Reply:
I'd still exclude it -- I think it's fragile and for MongoDB only. But, your call, I wouldn't fight hard.

Reply:
So exclude it and continue with writing a csuite test?

Reply:
No, I meant exclude it from the documentation -- sorry for the lack of clarity!